node/network/availability-recovery/src/lib.rs

```diff
@@ -521,9 +521,11 @@ async fn handle_signal(
 	match signal {
 		OverseerSignal::Conclude => Ok(true),
 		OverseerSignal::ActiveLeaves(ActiveLeavesUpdate { activated, .. }) => {
-			// if activated is non-empty, set state.live_block_hash to the first block in Activated.
-			if let Some(hash) = activated.get(0) {
-				state.live_block_hash = hash.0;
+			// if activated is non-empty, set state.live_block_hash to the highest block in `activated`
+			for activated in activated {
+				if activated.number > state.live_block_hash.0 {
+					state.live_block_hash = (activated.number, activated.hash)
```

Comment:
Keeping the name `live_block_hash` is slightly confusing. Maybe rename it to just `live_block`?